test/solidlsp/vue/test_vue_rename.py

```diff
@@ -0,0 +1,169 @@
+import os
+
+import pytest
+
+from solidlsp import SolidLanguageServer
+from solidlsp.ls_config import Language
+
+pytestmark = pytest.mark.vue
+
+
+class TestVueRename:
+    @pytest.mark.parametrize("language_server", [Language.VUE], indirect=True)
+    def test_rename_function_within_single_file(self, language_server: SolidLanguageServer) -> None:
+        file_path = os.path.join("src", "components", "CalculatorInput.vue")
+
+        symbols = language_server.request_document_symbols(file_path).get_all_symbols_and_roots()
+        handle_digit_symbol = next((s for s in symbols[0] if s.get("name") == "handleDigit"), None)
+
+        if not handle_digit_symbol or "selectionRange" not in handle_digit_symbol:
+            pytest.skip("handleDigit symbol not found - test fixture may need updating")
+
+        sel_start = handle_digit_symbol["selectionRange"]["start"]
+
+        workspace_edit = language_server.request_rename_symbol_edit(file_path, sel_start["line"], sel_start["character"], "processDigit")
+
+        assert workspace_edit is not None, "Should return WorkspaceEdit for rename operation"
+
+        has_changes = "changes" in workspace_edit and workspace_edit["changes"]
+        has_document_changes = "documentChanges" in workspace_edit and workspace_edit["documentChanges"]
+
+        assert has_changes or has_document_changes, "WorkspaceEdit should contain either 'changes' or 'documentChanges'"
+
+        if has_changes:
```

Comment:
should also handle the case of has_document_changes or remove the if - this whole block might never be called